codec-sockjs/src/main/java/io/netty/handler/codec/sockjs/handler/SessionHandler.java

```diff
@@ -0,0 +1,181 @@
+ * 2.0 (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.netty.handler.codec.sockjs.handler;
+
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.sockjs.SockJsSessionContext;
+import io.netty.handler.codec.sockjs.handler.SessionState.State;
+import io.netty.handler.codec.sockjs.protocol.CloseFrame;
+import io.netty.handler.codec.sockjs.protocol.MessageFrame;
+import io.netty.handler.codec.sockjs.protocol.OpenFrame;
+import io.netty.handler.codec.sockjs.util.Arguments;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+import static io.netty.util.ReferenceCountUtil.release;
+
+/**
+ * A ChannelHandler that manages SockJS sessions.
+ *
+ * For every connection received a new SessionHandler will be created and added to
+ * the pipeline.
+ *
+ * Depending on the type of connection (polling, streaming, send, or websocket)
+ * the type of {@link SessionState} that this session handles will differ.
+ *
+ */
+public class SessionHandler extends ChannelInboundHandlerAdapter {
+
+    public enum Event { CLOSE_SESSION, HANDLE_SESSION }
+
+    private static final InternalLogger logger = InternalLoggerFactory.getInstance(SessionHandler.class);
+    private static final OpenFrame OPEN_FRAME = new OpenFrame();
+    private static final CloseFrame GO_AWAY_FRAME = new CloseFrame(3000, "Go away!");
+
+    private final SessionState sessionState;
+
+    public SessionHandler(final SessionState sessionState) {
+        Arguments.checkNotNull(sessionState, "sessionState");
+        this.sessionState = sessionState;
+    }
+
+    @Override
+    public void channelRead(final ChannelHandlerContext ctx, final Object msg) throws Exception {
+        if (msg instanceof HttpRequest) {
+            if (logger.isDebugEnabled()) {
+                logger.debug("Handle session : {}", sessionState);
+            }
+            release(msg);
+            handleSession(ctx);
+        } else if (msg instanceof String) {
+            handleMessage((String) msg);
+        } else {
+            ctx.fireChannelRead(msg);
+        }
+    }
+
+    private void handleSession(final ChannelHandlerContext ctx) throws Exception {
+        switch (sessionState.getState()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)